app/feature/feature-travel-certificate/src/main/kotlin/com/hedvig/android/feature/travelcertificate/ui/TravelCertificateHistory.kt

```diff
@@ -0,0 +1,422 @@
+      HedvigScaffold(
+        navigateUp = navigateUp,
+        modifier = Modifier.clearFocusOnTap(),
+        topAppBarText = stringResource(id = R.string.PROFILE_ROW_TRAVEL_CERTIFICATE),
+        topAppBarActions = {
+          IconButton(
+            onClick = { showBottomSheet = true },
+            modifier = Modifier.size(40.dp),
+          ) {
+            Icon(
+              imageVector = Icons.Hedvig.Info,
+              contentDescription = stringResource(R.string.REFERRALS_INFO_BUTTON_CONTENT_DESCRIPTION),
+              modifier = Modifier.size(24.dp),
+            )
+          }
+        },
+      ) {
+        Spacer(modifier = Modifier.weight(1f))
+        HedvigInformationSection(
+          title = stringResource(id = R.string.something_went_wrong),
+          buttonText = stringResource(id = R.string.GENERAL_RETRY),
+          onButtonClick = {
+            reload()
+          },
+        )
+        Spacer(modifier = Modifier.weight(1f))
+      }
+    }
+
+    CertificateHistoryUiState.Loading -> {
+      Box(modifier = Modifier.fillMaxSize()) {
+        CircularProgressIndicator(modifier = Modifier.align(Alignment.Center))
+      }
+    }
+
+    is CertificateHistoryUiState.SuccessDownloadingCertificate -> {
+      LaunchedEffect(uiState.downLoadingUri) {
+        uiState.downLoadingUri?.let {
+          onShareTravelCertificate(it)
+        }
+      }
+    }
+
+    is CertificateHistoryUiState.SuccessDownloadingHistory -> {
+      ShowHistory(
+        onIconClick = { showBottomSheet = true },
+        onCertificateClick = onCertificateClick,
+        onContinue = onContinue,
+        navigateUp = navigateUp,
+        historyList = uiState.certificateHistoryList,
+      )
+    }
+  }
+}
+
+@Composable
+private fun ShowHistory(
+  onIconClick: () -> Unit,
+  onCertificateClick: (String) -> Unit,
+  onContinue: () -> Unit,
+  navigateUp: () -> Unit,
+  historyList: List<TravelCertificate>,
+) {
+  HedvigScaffold(
+    navigateUp = navigateUp,
+    modifier = Modifier.clearFocusOnTap(),
+    topAppBarText = stringResource(id = R.string.PROFILE_ROW_TRAVEL_CERTIFICATE),
+    topAppBarActions = {
+      IconButton(
+        onClick = { onIconClick() },
+        modifier = Modifier.size(40.dp),
+      ) {
+        Icon(
+          imageVector = Icons.Hedvig.Info,
+          contentDescription = stringResource(R.string.REFERRALS_INFO_BUTTON_CONTENT_DESCRIPTION),
+          modifier = Modifier.size(24.dp),
+        )
+      }
+    },
+  ) {
+    if (historyList.isEmpty()) {
+      Spacer(modifier = Modifier.weight(1f))
+      ShowInitialInfo()
+    } else {
+      ShowNotEmptyList(
+        list = historyList,
+        onCertificateClick = onCertificateClick,
+      )
+    }
+    Spacer(modifier = Modifier.weight(1f))
+    VectorInfoCard(
+      text = String.format(stringResource(id = R.string.travel_certificate_start_date_info), 45),
+      modifier = Modifier
+        .fillMaxWidth()
+        .padding(16.dp, 16.dp, 16.dp, 8.dp),
```

Comment:
Looking at this especially outside of the IDE with the hints can be a bit hard to understand about what is what.
I'd suggest this to be `.padding(horizontal = 16.dp)` and then for vertical spacings they can be spacers in-between the items themselves.

I very strongly recommend reading this https://www.jetpackcompose.app/articles/space-hoisting-should-I-use-the-padding-modifier-or-the-spacer-composable article around the philosophy of why using spacers is easier to use than padding on individual items, when the padding's goal is to make the items have some space between each other.